zebra-state/src/service/write.rs

```diff
@@ -285,3 +323,72 @@ pub fn write_blocks_from_channels(
     finalized_state.db.shutdown(true);
     std::mem::drop(finalized_state);
 }
+
+/// Log a message if this block was mined by Zebra.
+///
+/// Does not detect early Zebra blocks, and blocks with custom coinbase transactions.
+/// Rate-limited to every 1000 blocks using `last_zebra_mined_log_height`.
+fn log_if_mined_by_zebra(
+    tip_block: &ChainTipBlock,
+    last_zebra_mined_log_height: &mut Option<Height>,
+) {
+    // This logs at most every 2-3 checkpoints, which seems fine.
+    const LOG_RATE_LIMIT: u32 = 1000;
+
+    let height = tip_block.height.0;
+
+    if let Some(last_height) = last_zebra_mined_log_height {
+        if height < last_height.0 + LOG_RATE_LIMIT {
+            // If we logged in the last 1000 blocks, don't log anything now.
```

Comment:
Optional: a metrics counter would be nice here.

Reply:
Happy to do this in a follow-up PR to avoid conflicts.